enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,481 @@
+}
+
+type LocalVolumeDiscoverList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeDiscover `json:"items"`
+}
+
+type LocalVolumeDiscover struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeDiscoverSpec   `json:"spec"`
+	Status            LocalVolumeDiscoverStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeDiscoverSpec struct {
+	// NodeName is the name of the node where the LocalVolumeDiscover Daemon will be running
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// If specified tolerations is the list of toleration that is passed to the
+	// LocalVolumeDiscover Daemon
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type LocalVolumeDiscoverStatus struct {
+	Phase         string             `json:phase,omitempty`
+	NodeDeviceMap map[string]Devices `json:nodeDeviceMap,omitempty`
+}
+
+type Devices struct {
+	// Blockdevices will be containing the output of following command unmarshalled into the structured
+	// "lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint"
+	Blockdevices []Device `json:"blockdevices"`
+}
+
+type Device struct {
+	Name       string   `json:"name,omitempty"`
+	Model      string   `json:"model,omitempty"`
+	Vendor     string   `json:"vendor,omitempty"`
+	Kname      string   `json:"kname,omitempty"`
+	Pkname     string   `json:"pkname,omitempty"`
+	Type       string   `json:"type,omitempty"`
+	UUID       string   `json:"uuid,omitempty"`
+	Tran       string   `json:"tran,omitempty"`
+	Ro         string   `json:"ro,omitempty"`
+	Parttype   string   `json:"parttype,omitempty"`
+	Serial     string   `json:"serial,omitempty"`
+	Size       string   `json:"size,omitempty"`
+	Mountpoint string   `json:"mountpoint,omitempty"`
+	Rm         string   `json:"rm,omitempty"`
+	Children   []Device `json:children,omitempty`
+}
+```
+#### Worflow Chart
+![](https://i.imgur.com/FZGK1ki.png)
+
+#### Workflow of Device Discovery
+- The discovery of local devices will be done by the `LocalVolumeDiscover` CR. Once the LocalVolumeDiscover CR is deployed, the controller of the `LocalVolumeDiscover` CR will be exposing the available local devices of the node in the status of the LocalVolumeDiscover CR. These information will mostly be the output of 
+`lsblk -o name,model,vendor,kname,pkname,type,uuid,tran,ro,rm,parttype,serial,size,rota,mountpoint`
+
+Example of LocalVolumeDiscover CR after detection:
+```yaml
+apiVersion: local.storage.openshift.io/v1alpha1
+kind: LocalVolumeDiscover
+metadata:
+ name: example-volume-detect-1
+ namespace: localStorage
+spec:
+ nodeSelector:
+   nodeSelectorTerms:
+   - matchExpressions:
+     - key: kubernetes.io/hostname
+       operator: In
+       values:
+       - worker-0
+       - worker-1
+ tolerations:
+Status:
+ Phase: Discovered
+ NodeDeviceMap:
+   Worker-0:
```

Comment:
I was thinking since both `nodeSelectors` and `tolerations` will be the same for both LocalVolumeDiscover CR and AutoVolumeDetect CR so I kept it this way, still no hard resistance in  keeping CR per node.

Reply:
yeah we need just the `hostname` in that case.